ifrs17/Import/1ImportScope-Identities.ipynb

```diff
@@ -133,88 +141,25 @@
-                "\n                                                                                       ? new string[]{AocTypes.IA, AocTypes.CF, AocTypes.YCU, AocTypes.CRU, AocTypes.RCU} //Add IA, CF, YCU, CRU, RCU for in force",
-                "\n                                                                                       : new string[]{AocTypes.IA, AocTypes.CF, AocTypes.YCU}) //Add IA, CF, YCU,",
-                "\n                                                                                   .Select(aocType => new ImportIdentity {",
-                "\n                                                                                   AocType = aocType,",
-                "\n                                                                                   Novelty = n,",
-                "\n                                                                                   DataNode = Identity }))",
-                "\n                                                                                   .Concat(new ImportIdentity {",
-                "\n                                                                                   AocType = AocTypes.CF, //Add CF for Deferral",
-                "\n                                                                                   Novelty = Novelties.C,",
-                "\n                                                                                   DataNode = Identity",
-                "\n                                                                                   }.RepeatOnce())   ",
-                "\n                                                                                    .Concat(GetStorage().GetAllAocSteps(InputSource.Opening)",
-                "\n                                                                                              .Select(aocStep => new ImportIdentity {AocType = aocStep.AocType, ",
-                "\n                                                                                                 Novelty = aocStep.Novelty,",
-                "\n                                                                                                 DataNode = Identity",
-                "\n                                                                                                 }));   ",
-                "\n}",
+                "\n    private bool isReinsurance => GetStorage().DataNodeDataBySystemName[Identity].IsReinsurance;",
+                "\n    private IEnumerable<ImportIdentity> ParsedIdentities => GetScope<GetParsedAocSteps>(Identity).Values.Select(aocStep => new ImportIdentity {AocType = aocStep.AocType, Novelty = aocStep.Novelty, DataNode = Identity});",
+                "\n    private IEnumerable<string> rawVariableNovelties => GetStorage().GetRawVariables(Identity).Select(rv => rv.Novelty).Distinct().Concat(Novelties.C.RepeatOnce());",
```

Comment:
Also, how to you use them? Would it make sense to instead of a Distinct (which will keep this is an IEnumerable), do a .ToHashSet()?
You can keep the type as IEnumerable<string>, but underlying the compiler will know how to optimize the code, in case you use Contains an things like that.